### ISSUE TITLE
Introduce draft.js editor as input field for the previews.

### ISSUE THREAD

**Issue by omarreiss** [Yoast/yoast-components#111]
As a developer, I want to be able to have more control over the input elements in the different preview editors we have. draft.js will allow us to easily highlight or tag parts of our input in the editor. Step one is to replace the current inputs with draft.js. 

I don't know if draft.js also uses a native input under the hood of which we can easily determine the id through config. Otherwise we might need to add some functionality on top of draft.js to ensure that the input still gets exported somewhere.

**Comment by afercia** [comment]
First accessibility evaluation, based on the "rich" example. Starting from the basic things:

**Giving a name to all the things:**
- all the relevant UI parts need to be properly labeled to answer the question "what is this"? For example, the editor area itself is mapped to `role="textbox"` but doesn't have a label (label element, aria-label, whatever)
- same for the controls toolbars (all the bold, italic, etc.)
- the `role="textbox"` must be complemented with a aria-multiline property set to "true" to be correctly announced as the equivalent of a textarea; the editor is currently announced as a standard input field

I think this can be done pretty easily setting the Editor props.

**Keyboard accessibility:**
- all the controls must be operable with a keyboard, on the examples they're just `<span>` elements and they need to be changed in `<button >` elements
- this applies to the formatting controls and also to the add image, add link, etc.
- as a consequence, all the `onMouseDown` events must be changed in `onClick`

I've played a bit with this, seems it's pretty easy to do, and it works. Not sure why the original controls in the examples are built with `<span>` elements, or if there are specific technical reasons for that.

- focus style: should be added to all focusable controls
- keyboard shortcuts: see `getDefaultKeyBinding.js`; just some controls have shortcuts (bold, italic, etc.) not sure why other controls don't have any shortcuts, for example the controls for headings, ul, ol, blockquote. As far as I understand, it's possible to build custom key bindings.

**Screen readers:**
As always when `contenteditable` is used, accessibility for screen readers is very problematic. In most of the cases there's no report at all about formatting, i.e. it is not possible to understand when a text is bold, italic, a link... in a very few cases _some_ screen readers may announce bullet lists and a few other things.
To solve this issue, some offer an alternate, simplified editor (e.g. WordPress with the editor in text mode) while others have the resources to build a complete big application with a different approach, for example Google Docs.
Not sure what can be done here, maybe giving users a way to disable the editor and offer a simple textarea instead?

To recap:
- keyboard accessibility and semantics: I think it can be reasonably done
- screen readers: needs a decision
